tools/tcpretrans.py

```diff
@@ -187,11 +252,19 @@
     bpf_text = bpf_text.replace("IPV6_INIT", struct_init['ipv6']['count'])
     bpf_text = bpf_text.replace("IPV4_CORE", count_core_base.replace("COUNT_STRUCT", 'ipv4_count'))
     bpf_text = bpf_text.replace("IPV6_CORE", count_core_base.replace("COUNT_STRUCT", 'ipv6_count'))
+    if BPF.tracepoint_exists("tcp", "tcp_retransmit_skb"):
+        bpf_text_tracepoint = bpf_text_tracepoint.replace("IPV4_CODE", struct_init_tracepoint['ipv4']['count'])
+        bpf_text_tracepoint = bpf_text_tracepoint.replace("IPV6_CODE", struct_init_tracepoint['ipv6']['count'])
+        bpf_text += bpf_text_tracepoint
```

Comment:
Looks good in general. Could you do a little bit refactoring not to load `trace_retransmit` function if tracepoint exists? If would be confusing if both `trace_retransmit` and the tracepoint program are loaded.

Reply:
Okay. Either kprobe or tracepoint will be loaded, but not both, given '-l' is not there.